### PR TITLE
Add feature flag for "data permission" options in account

### PR DESCRIPTION
**Pull request by marcoconti83**
# Reason for this pull request

Need to add feature flags for some options in the data permission settings, namely:
- Send anonymous crash reports
- Allow marketing communication

We want to hide those options completely when the feature flag is removed.

# Changes

Removed the to individual options if the flags are false, and also removed the main "data permission" setting button if both flags are false, as there is no point in having an empty section.

# Testing
See the issue with testing in https://github.com/wireapp/wire-android/pull/1964
#### APK
[Download build #12292](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12292/artifact/build/artifact/wire-dev-PR1967-12292.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala [hunk 1]
```diff
@@ -46,42 +48,46 @@ class DataUsagePermissionsView(context: Context, attrs: AttributeSet, style: Int
...
     }
   }
 
   val receiveNewsAndOffersSwitch = returning(findById[SwitchPreference](R.id.preferences_receive_news_and_offers)) { v =>
 
-    def setReceivingNewsAndOffersSwitchEnabled(enabled: Boolean) = {
-      v.setEnabled(enabled)
-      v.setAlpha(if (enabled) 1.0f else 0.5f)
-    }
+    if (BuildConfig.ALLOW_MARKETING_COMMUNICATION) {
```
Comment (marcoconti83) [inline app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala]: Here I just added an `if`. This causes all indentation to change so all following lines look changed, but I did not modify the behavior, just indented.

File: app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala [hunk 2]
```diff
@@ -46,42 +48,46 @@ class DataUsagePermissionsView(context: Context, attrs: AttributeSet, style: Int
...
-    def setAnalyticsSwitchEnabled(enabled: Boolean) = {
-      v.setEnabled(enabled)
-      v.setAlpha(if (enabled) 1.0f else 0.5f)
-    }
-
-    v.setPreference(analyticsPrefKey, global = true)
-    v.pref.flatMap(_.signal).onChanged {
-      case true  => tracking.optIn()
-      case false => tracking.optOut()
+    if (BuildConfig.SUBMIT_CRASH_REPORTS) {
```
Comment (marcoconti83) [inline app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala]: The `setAnalyticsSwitchEnabled` was not previously used at all so I removed it

File: app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala [hunk 3]
```diff
@@ -46,42 +48,46 @@ class DataUsagePermissionsView(context: Context, attrs: AttributeSet, style: Int
...
-              ContextUtils.showGenericErrorDialog()
+      setReceivingNewsAndOffersSwitchEnabled(false)
+      am.head.flatMap(_.hasMarketingConsent).foreach { isSet =>
+        v.setChecked(isSet, disableListener = true)
+        setReceivingNewsAndOffersSwitchEnabled(true)
+
+        v.onCheckedChange.onUi { set =>
+          setReceivingNewsAndOffersSwitchEnabled(false)
+          am.head.flatMap(_.setMarketingConsent(Some(set))).foreach { resp =>
+            resp match {
```
Comment (yeryomenkom) [inline app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala]: Do not use pattern matching if you can avoid it.
In this case do 
if (resp.isLeft)

File: app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala [hunk 4]
```diff
@@ -23,6 +23,7 @@ import android.os.Bundle
 import android.util.AttributeSet
 import android.view.View
 import android.widget.LinearLayout
+import com.waz.ZLog
```
[the remark this answers is not in the document]
Comment (marcoconti83) [reply app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala]: Fixed

File: app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala (hunk 3 above)
Comment (marcoconti83) [reply app/src/main/scala/com/waz/zclient/preferences/pages/DataUsagePermissionsView.scala]: Fixed

**Pull request merged**